packages/graphql/src/schema-model/annotation/Annotation.ts

```diff
@@ -17,140 +17,115 @@
+    customResolver: CustomResolverAnnotation;
+    jwt: JWTPayloadAnnotation;
+    selectable: SelectableAnnotation;
+    populatedBy: PopulatedByAnnotation;
+    subscription: SubscriptionAnnotation;
+    authorization: AuthorizationAnnotation;
+    default: DefaultAnnotation;
+    settable: SettableAnnotation;
+    cypher: CypherAnnotation;
+    fulltext: FullTextAnnotation;
+    limit: LimitAnnotation;
+    id: IDAnnotation;
+    key: KeyAnnotation;
+    authentication: AuthenticationAnnotation;
+    timestamp: TimestampAnnotation;
+    filterable: FilterableAnnotation;
+    jwtClaim: JWTClaimAnnotation;
+    query: QueryAnnotation;
+    coalesce: CoalesceAnnotation;
+    subscriptionsAuthorization: SubscriptionsAuthorizationAnnotation;
+    mutation: MutationAnnotation;
+    relayId: RelayIDAnnotation;
+    unique: UniqueAnnotation;
 };
 
-export function annotationToKey(ann: Annotation): keyof Annotations {
-    if (ann instanceof CypherAnnotation) return AnnotationsKey.cypher;
-    if (ann instanceof AuthorizationAnnotation) return AnnotationsKey.authorization;
-    if (ann instanceof AuthenticationAnnotation) return AnnotationsKey.authentication;
-    if (ann instanceof KeyAnnotation) return AnnotationsKey.key;
-    if (ann instanceof SubscriptionsAuthorizationAnnotation) return AnnotationsKey.subscriptionsAuthorization;
-    if (ann instanceof LimitAnnotation) return AnnotationsKey.limit;
-    if (ann instanceof DefaultAnnotation) return AnnotationsKey.default;
-    if (ann instanceof CoalesceAnnotation) return AnnotationsKey.coalesce;
-    if (ann instanceof CustomResolverAnnotation) return AnnotationsKey.customResolver;
-    if (ann instanceof IDAnnotation) return AnnotationsKey.id;
-    if (ann instanceof MutationAnnotation) return AnnotationsKey.mutation;
-    if (ann instanceof PluralAnnotation) return AnnotationsKey.plural;
-    if (ann instanceof FilterableAnnotation) return AnnotationsKey.filterable;
-    if (ann instanceof FullTextAnnotation) return AnnotationsKey.fulltext;
-    if (ann instanceof PopulatedByAnnotation) return AnnotationsKey.populatedBy;
-    if (ann instanceof QueryAnnotation) return AnnotationsKey.query;
-    if (ann instanceof PrivateAnnotation) return AnnotationsKey.private;
-    if (ann instanceof SelectableAnnotation) return AnnotationsKey.selectable;
-    if (ann instanceof SettableAnnotation) return AnnotationsKey.settable;
-    if (ann instanceof TimestampAnnotation) return AnnotationsKey.timestamp;
-    if (ann instanceof UniqueAnnotation) return AnnotationsKey.unique;
-    if (ann instanceof SubscriptionAnnotation) return AnnotationsKey.subscription;
-    if (ann instanceof JWTClaimAnnotation) return AnnotationsKey.jwtClaim;
-    if (ann instanceof JWTPayloadAnnotation) return AnnotationsKey.jwtPayload;
-    if (ann instanceof RelayIDAnnotation) return AnnotationsKey.relayId;
-    throw new Error("annotation not known");
-}
+export type AnnotationFactory<T extends Annotation> = (
```

Comment:
Maybe `AnnotationParser` is a better type for this? Just to keep the same naming, no other reason